packages/testing/scripts/fetchMetadata.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable */
+
+const { HttpProvider, WsProvider, ApiPromise } = require('@polkadot/api')
+const yargs = require('yargs')
+const fs = require('fs')
+
+const { argv } = yargs
+  .option('format', {
+    alias: 'f',
+    description: 'output format: {hex, json}',
+    type: 'string',
+    default: 'json',
+    choices: ['hex', 'json'],
+  })
+  .option('endpoint', {
+    alias: 'e',
+    description: 'http or ws endpoint from which to fetch metadata',
+    type: 'string',
+    demandOption: true,
+  })
+  .option('outfile', {
+    alias: 'o',
+    description: 'path to output file',
+    type: 'string',
+    demandOption: true,
+  })
+  .help()
+  .alias('help', 'h')
+
+let provider
+
+switch (true) {
+  case argv.endpoint.startsWith('http'):
+    provider = new HttpProvider(argv.endpoint)
+    break
+  case argv.endpoint.startsWith('ws'):
+    provider = new WsProvider(argv.endpoint, false)
+    break
+  default:
+    throw new Error(
+      `can only handle ws/wss and http/https endpoints, received ${argv.endpoint}`
+    )
+}
+
+const api = new ApiPromise({ provider })
+
+let exitCode
+
+async function fetch() {
+  await api.isReady
+  let metadata
+  switch (argv.format) {
+    case 'hex':
+      metadata = `"${api.runtimeMetadata.toHex()}"`
+      break
+    case 'json':
+      metadata = JSON.stringify(api.runtimeMetadata.toJSON(), null, 2)
+      break
+    default:
+      throw new Error('unexpected output format')
+  }
+
+  console.log(
+    `writing metadata to ${argv.outfile}:\n${metadata.substring(0, 100)}...`
+  )
+  fs.writeFileSync(argv.outfile, metadata)
+  console.log('success')
+  exitCode = 0
+}
+
+const timeout = new Promise((_, reject) => {
+  setTimeout(() => {
+    exitCode = exitCode || 124
+    reject(new Error('timeout waiting for metadata fetch'))
+  }, 10000)
+})
+
+Promise.race([fetch(), timeout])
```

Comment:
Ha! interesting way of doing a timeout

Reply:
Is it so unusual? I thought that's how it's done in js. What would you do?

Reply:
Another thing I'm realising now: do we need to cancel timeouts? I had the feeling that the script wouldn't exit until the timeout timer has elapsed. And jest has started indicating in it's "process has not exited 1s after end of tests" message that timeouts might be a cause.

Reply:
It's just cool to let them `race`, that's why I found it interesintg. Normally you might have them set-up sequentially, because you might want to cleanup the fetch, when the timeout triggers. But since we exit in this case, that should be fine.